browser/extensions/brave_crx_generation_browsertest.cc

```diff
@@ -0,0 +1,152 @@
+
+#include "base/command_line.h"
+#include "base/files/file_util.h"
+#include "base/files/scoped_file.h"
+#include "base/path_service.h"
+#include "brave/common/brave_paths.h"
+#include "chrome/browser/extensions/crx_installer.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "components/crx_file/crx_verifier.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/test_utils.h"
+#include "crypto/rsa_private_key.h"
+#include "crypto/sha2.h"
+#include "extensions/browser/crx_file_info.h"
+#include "extensions/browser/extension_creator.h"
+#include "extensions/browser/extension_system.h"
+#include "extensions/browser/notification_types.h"
+#include "extensions/common/extension.h"
+
+namespace extensions {
+
+class BraveCrxGenerationTest : public InProcessBrowserTest {
+ public:
+  BraveCrxGenerationTest() {
+    CHECK(temp_directory_.CreateUniqueTempDir());
+    brave::RegisterPathProvider();
+  }
+
+  const Extension* InstallExtension(const base::FilePath& crx_path,
+                                    crx_file::VerifierFormat format) {
+    auto installer = CrxInstaller::CreateSilent(
+        ExtensionSystem::Get(browser()->profile())->extension_service());
+    installer->set_allow_silent_install(true);
+    installer->set_install_cause(extension_misc::INSTALL_CAUSE_USER_DOWNLOAD);
+    installer->set_creation_flags(Extension::FROM_WEBSTORE);
+
+    content::WindowedNotificationObserver observer(
+        NOTIFICATION_CRX_INSTALLER_DONE,
+        content::NotificationService::AllSources());
+    installer->InstallCrxFile(CRXFileInfo(crx_path, format));
+    observer.Wait();
+    content::Details<const Extension> details = observer.details();
+    return details.ptr();
+  }
+  base::FilePath GetTestDataDir() {
+    base::FilePath test_data_dir;
+    base::ScopedAllowBlockingForTesting allow_blocking;
+    CHECK(base::PathService::Get(brave::DIR_TEST_DATA, &test_data_dir));
+    return test_data_dir;
+  }
+
+  base::FilePath CreateTestCrx() {
+    base::ScopedAllowBlockingForTesting allow_blocking;
+
+    brave::RegisterPathProvider();
+    const auto extension_dir =
+        GetTestDataDir().AppendASCII("extensions/trivial_extension");
+
+    const auto private_key_path =
+        GetTestDataDir().AppendASCII("extensions/trivial_extension.pem");
+    const auto crx_path = temp_directory_.GetPath().AppendASCII("test.crx");
+
+    ExtensionCreator creator;
+    if (!creator.Run(extension_dir, crx_path, private_key_path,
+                     base::FilePath(), ExtensionCreator::kOverwriteCRX)) {
+      LOG(ERROR) << creator.error_message();
+      return base::FilePath();
+    }
+    return crx_path;
+  }
+
+  base::ScopedTempDir temp_directory_;
+};
+
+// Generate a publish key hash from .pem file in the format used in
+// crx_verifier.cc.
+std::vector<uint8_t> GetPublicKeyHash(const base::FilePath& pem_path) {
```

Comment:
put it into fixture or anonymous namespace.

Reply:
Done